diff --git a/90-vpdupdate.rules b/90-vpdupdate.rules
new file mode 100644
index 0000000..7c4f9cc
--- /dev/null
+++ b/90-vpdupdate.rules
@@ -0,0 +1 @@
+KERNELS=="*", ACTION=="*", RUN+="/bin/touch /var/lib/lsvpd/run.vpdupdate"
diff --git a/Makefile.am b/Makefile.am
index 592603b..d4ec377 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -24,7 +24,7 @@
 
 INCLUDES = -I./src/
 
-EXTRA_DIST = bootstrap.sh
+EXTRA_DIST = bootstrap.sh 90-vpdupdate.rules run.vpdupdate
 libvpd_la_SOURCES = src/vpdretriever_c.c \
 		src/common.c \
 		src/vpddbenv_c.c \
@@ -60,4 +60,12 @@
 
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 libtool: $(LIBTOOL_DEPS)
-	$(SHELL) ./config.status --recheck
+	$(SHELL) ./config.status --recheck+
+
+install-exec-hook:
+	mkdir -p $(DESTDIR)/${sysconfdir}/udev/rules.d/
+	mkdir -p $(DESTDIR)/${localstatedir}/lib/lsvpd/
+	install -D --mode=644 run.vpdupdate \
+		${DESTDIR}/${localstatedir}/lib/lsvpd/run.vpdupdate
+	install -D --mode=644 90-vpdupdate.rules \
+                ${DESTDIR}/${sysconfdir}/udev/rules.d/90-vpdupdate.rules
--- a/libvpd.spec.in
+++ b/libvpd.spec.in
@@ -50,6 +50,8 @@ Contains header files for building with libvpd.
 %exclude %{_libdir}/*.la
 %{_libdir}/libvpd_cxx-@GENERIC_RELEASE@.so.*
 %{_libdir}/libvpd-@GENERIC_RELEASE@.so.*
+%{_sysconfdir}/udev/rules.d/90-vpdupdate.rules
+%{_var}/lib/lsvpd/run.vpdupdate
 
 %files devel
 %defattr(-,root,root,-)
@@ -62,6 +64,9 @@ Contains header files for building with libvpd.
 
 %changelog
 
+* Thu Oct 31 2013 Phani Yadav <phayadav@linux.vnet.ibm.com>
+- Automation of vpdupdate
+
 * Wed Feb 20 2013 Harsh P Bora <harsh@linux.vnet.ibm.com> 2.2.1-1
 - Added Vendor tag in RPM package.
 
diff --git a/run.vpdupdate b/run.vpdupdate
new file mode 100644
index 0000000..e69de29
diff --git a/src/libvpd-2/vpdretriever.hpp b/src/libvpd-2/vpdretriever.hpp
index 57d3378..0d91ac4 100644
--- a/src/libvpd-2/vpdretriever.hpp
+++ b/src/libvpd-2/vpdretriever.hpp
@@ -42,6 +42,7 @@ namespace lsvpd
 		public:
 			static const string DEFAULT_DIR;
 			static const string DEFAULT_FILE;
+			static const string UDEV_NOTIFY_FILE;
 			
 			/**
 			 * Builds A VpdRetriever object that can be used for reading the
diff --git a/src/vpdretriever.cpp b/src/vpdretriever.cpp
index edfac22..95090fd 100644
--- a/src/vpdretriever.cpp
+++ b/src/vpdretriever.cpp
@@ -27,6 +27,11 @@
 
 #include <vector>
 #include <string>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <cstdlib>
+#include <errno.h>
 
 using namespace std;
 
@@ -34,6 +39,7 @@ namespace lsvpd
 {
 	const string VpdRetriever::DEFAULT_DIR  ( "/var/lib/lsvpd/" );
 	const string VpdRetriever::DEFAULT_FILE ( "vpd.db" );
+	const string VpdRetriever::UDEV_NOTIFY_FILE ( "run.vpdupdate" );
 
 	VpdRetriever::VpdRetriever( string envDir,
 		string dbFileName ) throw( VpdException& )
@@ -51,6 +57,40 @@ namespace lsvpd
 	
 	VpdRetriever::VpdRetriever( ) throw( VpdException& )
 	{
+		struct stat vpd_stat,udev_stat;
+		const string vpddb = VpdRetriever::DEFAULT_DIR + VpdRetriever::DEFAULT_FILE;
+		const string udev_file = VpdRetriever::DEFAULT_DIR + VpdRetriever::UDEV_NOTIFY_FILE;
+		Logger logger;
+		int flag = 1;
+
+		/* Check if stat is successful for UDEV_NOTIFY_FILE. */
+		if ( stat ( udev_file.c_str(), &udev_stat ) != 0 )
+		{
+			logger.log( "libvpd: Unable to stat udev rule file, run.vpdupdate.", LOG_INFO );
+		}
+		else
+		{
+			/* Find the modification time for default DB. */
+			if ( stat ( vpddb.c_str(), &vpd_stat ) != 0 )
+			{
+				logger.log( "libvpd: Unable to stat vpd.db file.", LOG_INFO );
+				if ( errno == ENOENT )
+					vpd_stat.st_mtime = 0;
+				else
+					flag = 0;
+			}
+
+			if ( flag && ( udev_stat.st_mtime > vpd_stat.st_mtime ) )
+			{
+				/*
+				 * This implies there were changes to devices on the system
+				 * after the VPD db was created/modified. So
+				 * run vpdupdate.
+				 */
+				logger.log( "libvpd: Running vpdupdate to update the default db.", LOG_INFO );
+				system( "/sbin/vpdupdate >/dev/null 2>&1" );
+			}
+		}
 		db = new VpdDbEnv( VpdRetriever::DEFAULT_DIR,
 					VpdRetriever::DEFAULT_FILE, true );
 
